### DISCUSSION TITLE
On generalisation error

### DISCUSSION THREAD

**Discussion by tony-teng-yt377** [Part-II-Computational-Physics/.github#79] [General]
Now that I know the generalisation is the difference between loss function for training and test data, I wonder whether we should compare the loss function at each step (do we compare the loss function given by trained model (only one values) on test set to the loss function on training data set at each step (100 values)? ) Or do we just care about this difference at the end?

**Comment by PatrickC12** [comment]
![image](https://user-images.githubusercontent.com/123903514/225418804-51ab1b33-d319-4726-9a05-d8406b14b946.png)

I'm not sure, but this formula seems to imply we just average over the losses on the test data, having previously trained the NN on the training set.

**Reply by tony-teng-yt377** [reply]
Does that mean that we just obtain a single value for generalisation value by applying the trained model (after 100 steps) on test set and training set and compute their difference?

**Reply by tony-teng-yt377** [reply]
Feel like it's just quite unclear about what we need to do even after knowing this definition

**Reply by PatrickC12** [reply]
So I think we initialise the NN, train it with the training set. Then feed the test data to the NN without reinitialising it. After that just average the losses list that is created by the optimiser.

**Reply by ceciliegl** [reply]
Sorry for being unclear. You are asked to compute the generalisation error after training the neural network, i.e. only a single value for each optimiser.

However, if you like to compute it at each step (as I see you have already done in a previous post), that is of course fine too. You are very welcome to include that in your answer and discussion, if you'd like!

**Reply by tony-teng-yt377** [reply]
Ok now I get it, thank you very much!

**Reply by PatrickC12** [reply]
So is the generalisation error the difference between the two losses, or just the losses of the NN acting on the test data ? and if the 2nd, is it just the average of the losses or the loss after the final step ?

**Reply by tony-teng-yt377** [reply]
I think it's the difference between two losses, because you would just want to see how good this model generalise to an unseen data set

**Reply by PatrickC12** [reply]
But wasn't the original data set also previously unseen by the model?

**Reply by tony-teng-yt377** [reply]
Yes, but the idea is that the model might create some zigzags specific to this one set of random data, which does not perform as well if we give another set. So testing it on the uniform circle tells us how much those zigzag will influence the performance if we use this model on another randomly generated data. That's at least how I understood it.

**Reply by PatrickC12** [reply]
ah ok, just confused since wikipedia seems to give different way of calculating generelisation error compared to the lecture notes.

**Reply by ceciliegl** [reply]
Yes, it is the difference between the two losses😊
